File: Domain/Shift.cs
using System;

namespace Domain
{
    public class Shift
    {
        public Guid Id { get; set; }
        public DateTime Start { get; set; }
        public DateTime End { get; set; }
        public Guid LicenseId { get; set; }
        public License License { get; set; }
        public Guid LocationId { get; set; }
        public Location Location { get; set; }
        public Guid RoomId { get; set; }
        public Room Room { get; set; }
        public Guid TechnologistId { get; set; }
        public Technologist Technologist { get; set; }
        public Guid ModalityId { get; set; }
        public Modality Modality { get; set; }

    }
}

File: client-app/src/app/models/shift.ts
export interface IShift {
  id: string;
  start: Date;
  end: Date;
  licenseId: string;
  licenseDisplayName: string;
  locationId: string;
  locationName: string;
  roomId: string;
  roomName: string;
  technologistId: string;
  technologistInitial: string;
  modalityId: string;
}

export interface IShiftFormValues extends Partial<IShift> {
  date?: Date;
}

export class ShiftFormValues implements IShiftFormValues {
  id?: string = undefined;
  date?: Date = undefined;
  start?: Date = undefined;
  end?: Date = undefined;
  licenseId: string = "";
  locationId: string = "";
  roomId: string = "";
  technologistId: string = "";
  modalityId: string = "";

  constructor(init?: ShiftFormValues) {
    if (init && !init.date) {
      init.date = init.start;
    }
    Object.assign(this, init);
  }
}
